pennylane/devices/default_qubit.py

```diff
@@ -79,32 +78,57 @@
     from pennylane.operation import Operator
 
 
-def stopping_condition(op: Operator) -> bool:
+_special_operator_support = {
+    "QFT": lambda op: len(op.wires) < 6,
+    "GroverOperator": lambda op: len(op.wires) < 13,
+    "FromBloq": lambda op: len(op.wires) < 4 and op.has_matrix,
+    "Snapshot": lambda _: True,
+    "Allocate": lambda _: True,
+    "Deallocate": lambda _: True,
+}
+"""Map from gates with a special support condition."""
+
+
+class _Partial:
```

Comment:
What's the purpose of this?

Reply:
Being able to compare for testing. `partial(f, a =1) != partial(f, a=1)`. So special `_Partial` with that equality.